tests/ui/array-slice-vec/bounds-check-no-overflow.rs

```diff
@@ -1,6 +1,6 @@
 //@ run-fail
 //@ error-pattern:index out of bounds
-//@ ignore-emscripten no processes
+//@ needs-subprocess
```

Comment:
This is now ignored on sgx and wasm, too. 🤔

Reply:
I'll double-check this one, seems a bit sus

Reply:
I double-checked this one, AFAICT this is correct, because the index operation is expected to panic (the test is `//@ run-fail`) and so would indeed need subprocess support. As to why this did not previously have `ignore-sgx` or `ignore-wasm`:

- We don't run `sgx` tests in our CI, and instead sgx maintainers are expected to report back and fix tests that fail in their testing environments.
- `ignore-wasm` is actually implied by `ignore-emscripten` because of this beautiful clause

    https://github.com/rust-lang/rust/blob/1c9837df1dde9b234229709e89b3672bd3cf04a4/src/tools/compiletest/src/header/cfg.rs#L108-L117

    that I kinda want to rip out

Reply:
Thanks for that analysis! That makes further reviewing much easier